src/util/variant.h

```diff
@@ -0,0 +1,18 @@
+/*******************************************************************\
+
+Module: typedef for optional class template. To be replaced with
+        std::optional once C++17 support is enabled
```

Comment:
Should this be in 'nonstd/' with the existing `optional` code?

Reply:
Since it is a submodule, I don't think I can change the namespace it is in? Or do you mean, making the `using nostd::variantt = ...` - the `optionalt` is declared in the global scope so I figured this was more consistent. 

 But your comment highlighted a copy-pasta error in the docs - will fix.

Reply:
Thinking about it; just a comment saying ```// Swap for std::variant when switching to C++17``` or something.

Reply:
Not sure I follow, comment currently reads (typo not withstanding):

>  To be replaced with std::variant once C++17 support is enabled

Is your suggestion your wording is clearer? Happy to change if that is indeed what you mean, but for me they are equivalent